docs/source/plugin/privileges.rst

```diff
@@ -0,0 +1,172 @@
+===============
+User privileges
+===============
+
+IRC users can have privileges in a **channel**, given by MODE messages such as:
+
+.. code-block:: irc
+
+    MODE #example +ov Nickname Nickname
+
+This will give both OP and Voice privileges to the user named "Nickname" in the
+"#example" channel (and only in this channel). When Sopel receives a MODE
+message it registers and updates its knowledge of a user's privileges in a
+channel, which can be used by plugins in various way.
+
+Access rights
+=============
+
+Privileged users
+----------------
+
+A plugin can limit who can trigger its callables using the
+:func:`~sopel.plugin.require_privilege` decorator::
+
+    from sopel import plugin, privileges
+
+    @plugin.require_privilege(privileges.OP)
+    @plugin.require_chanmsg
+    @plugin.command('chanopcommand')
+    def chanop_command(bot, trigger):
+        # only a channel operator can use this command
+
+This way, only users with OP privileges or above in a channel can use the
+command ``chanopcommand`` in that channel: other users will be ignored by the
+bot. It is possible to tell these users why with the ``message`` parameter::
+
+    @plugin.require_privilege(privileges.OP, 'You need +o privileges.')
+
+.. note::
+
+    A command that requires channel privileges will probably be used for
+    channels only. For that purpose, you should use the
+    :func:`sopel.plugin.require_chanmsg` decorator.
```

Comment:
This is more important than you might think: `require_privilege` behaves as if the given minimum privilege level is always met if the message isn't from a channel. I'd go stronger here:

```suggestion
.. important::

    A command that requires channel privileges will always execute if
    called from a private message to the bot. You can use the
    :func:`sopel.plugin.require_chanmsg` decorator to ignore the command
    if it's called in PMs.
```